python/ray/tune/utils/mock.py

```diff
@@ -108,14 +109,22 @@ class FailureInjectorCallback(Callback):
     def __init__(self,
                  config_path="~/ray_bootstrap_config.yaml",
                  probability=0.1,
+                 time_between_checks=0,
```

Comment:
Because `on_step_begin` is called much more often with placement groups, we need to introduce some time-based checks here